uint8[] buffer
uint32 maxLen
---
uint32 totalReceived